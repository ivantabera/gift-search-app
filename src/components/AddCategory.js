import React, {useState} from 'react';
import PropTypes from 'prop-types';

//Se destructuran los argumentos del componente 
export const AddCategory = ( { setCategories } ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        //Validacion para evitar que el input tenga menos de 2 letras
        if ( inputValue.trim().length > 2 ) {
            // mandamos la referencia con el operador de arrastre a setCategories y el texto en el input con la variable (inputValue) para que haga la insercion
            setCategories( cate => [ inputValue, ...cate]);
            // borrar el valor del input para evitar el doble posteo
            setInputValue('');
        }

        
    }

    return (
        <form onSubmit = {handleSubmit}>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}