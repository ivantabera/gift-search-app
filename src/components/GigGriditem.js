import React from 'react'

export const GigGriditem = ({ title, url }) => {
    return (
        <div>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
