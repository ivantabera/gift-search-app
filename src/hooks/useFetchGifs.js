import { useState, useEffect} from "react"
import { getGif } from "../helpers/GetGif";

export const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data:[],
        loading:true
    });

    
    // Declaramos la funcion getGif dentro del useEffect() con [] para que solo se ejecute 1 ves cuando cargue la pagina
    // El Hook de efecto te permite llevar a cabo efectos secundarios en componentes funcionales:
    useEffect(() => {
        
        getGif(category)
            .then( imgs => {
                
                setState({
                    data:imgs,
                    loading:false
                });
                
            })

    }, [category])


    return state; // {data:[], loading:true}
}