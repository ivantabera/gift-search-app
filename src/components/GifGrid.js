import React, { useState, useEffect } from 'react'
import { GigGriditem } from './GigGriditem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    // Declaramos la funcion getGif dentro del useEffect() para que solo se ejecute 1 ves cuando cargue la pagina
    // El Hook de efecto te permite llevar a cabo efectos secundarios en componentes funcionales:
    useEffect( () => {
        getGif();
    }, []);

    // haremos una funcion asincrona para recibir los datos de la API
    const getGif = async() => {

        // Guardamos nuestra url del api en una constante
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick and Morty&limit=10&api_key=fb36rhWkdzSVD5NPLfFas2Wryuq82AM2';
        // hacemos la peticion con un fetch
        const resp = await fetch( url );
        // desestructuramos la data para que solo nos de la data y evitar la demas informacion
        const {data} = await resp.json();

        // Extraemos la informacion barriendo cada imagen
        const gifs = data.map( img => {
            // el return es el que transforma cada elemento del arreglo img
            return{
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs)
        setImages(gifs);
    }

    // gitGif();

    return (
        <>
            <h3> { category } </h3>
            <div className="card-grid">
                {
                    images.map( ( img ) => {
                        return <GigGriditem 
                                    key = { img.id }
                                    { ...img }
                                />
                    })
                };
            </div>
        </>
    )
}
