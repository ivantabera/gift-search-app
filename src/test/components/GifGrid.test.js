import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en GifGrid', () => {

    let category = "Hola mundo";
    
    test('Debe de mostrar el componente de manera correcta', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        let wrapper = shallow( <GifGrid category={category} />);
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de mostrar items cuando se cargan imagentes con el useFetch', () => {
        const gifs = [{
            id:'ASB',
            url:'https://welcome.com',
            title:'Hola mundo'
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });
        const wrapper = shallow( <GifGrid category={category} />);
        expect( wrapper ).toMatchSnapshot();
    })
    

})
