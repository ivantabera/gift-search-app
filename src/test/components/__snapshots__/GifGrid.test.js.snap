// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe de mostrar el componente de manera correcta 1`] = `
<Fragment>
  <h3>
     
    Hola mundo
     
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en GifGrid Debe de mostrar items cuando se cargan imagentes con el useFetch 1`] = `
<Fragment>
  <h3>
     
    Hola mundo
     
  </h3>
  <div
    className="card-grid"
  >
    <GigGriditem
      id="ASB"
      key="ASB"
      title="Hola mundo"
      url="https://welcome.com"
    />
  </div>
</Fragment>
`;
